posthog/clickhouse/migrations/0158_sessions_v3_remake.py

```diff
@@ -0,0 +1,25 @@
+from posthog.clickhouse.client.connection import NodeRole
+from posthog.clickhouse.client.migration_tools import run_sql_with_exceptions
+from posthog.models.raw_sessions.sql_v3 import (
+    DISTRIBUTED_RAW_SESSIONS_TABLE_SQL_V3,
+    DROP_RAW_SESSION_DISTRIBUTED_TABLE_SQL_V3,
+    DROP_RAW_SESSION_SHARDED_TABLE_SQL_V3,
+    DROP_RAW_SESSION_VIEW_SQL_V3,
+    DROP_RAW_SESSION_WRITABLE_TABLE_SQL_V3,
+    RAW_SESSIONS_CREATE_OR_REPLACE_VIEW_SQL_V3,
+    SHARDED_RAW_SESSIONS_TABLE_SQL_V3,
+    WRITABLE_RAW_SESSIONS_TABLE_SQL_V3,
+)
+
+operations = [
+    run_sql_with_exceptions(DROP_RAW_SESSION_VIEW_SQL_V3()),
+    run_sql_with_exceptions(DROP_RAW_SESSION_SHARDED_TABLE_SQL_V3()),
+    run_sql_with_exceptions(
+        DROP_RAW_SESSION_DISTRIBUTED_TABLE_SQL_V3(), node_roles=[NodeRole.DATA, NodeRole.COORDINATOR]
+    ),
+    run_sql_with_exceptions(DROP_RAW_SESSION_WRITABLE_TABLE_SQL_V3()),
+    run_sql_with_exceptions(WRITABLE_RAW_SESSIONS_TABLE_SQL_V3()),
+    run_sql_with_exceptions(DISTRIBUTED_RAW_SESSIONS_TABLE_SQL_V3(), node_roles=[NodeRole.DATA, NodeRole.COORDINATOR]),
+    run_sql_with_exceptions(SHARDED_RAW_SESSIONS_TABLE_SQL_V3()),
```

Comment:
SHARDED_RAW_SESSIONS_TABLE_SQL_V3 must be executed on `node_roles=NodeRole.DATA`